python/plugins/processing/algs/qgis/SimplifyGeometries.py

```diff
@@ -58,20 +59,21 @@ def group(self):
 
     def __init__(self):
         super().__init__()
-        self.addParameter(ParameterVector(self.INPUT,
-                                          self.tr('Input layer'),
-                                          [dataobjects.TYPE_VECTOR_POLYGON, dataobjects.TYPE_VECTOR_LINE]))
+        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
+                                                              self.tr('Input layer'),
+                                                              [dataobjects.TYPE_VECTOR_POLYGON, dataobjects.TYPE_VECTOR_LINE]))
         self.methods = [self.tr('Distance (Douglas-Peucker)'),
                         'Snap to grid',
                         'Area (Visvalingam)']
-        self.addParameter(ParameterSelection(
+        self.addParameter(QgsProcessingParameterEnum(
             self.METHOD,
             self.tr('Simplification method'),
-            self.methods, default=0))
-        self.addParameter(ParameterNumber(self.TOLERANCE,
-                                          self.tr('Tolerance'), 0.0, 10000000.0, 1.0))
+            self.methods, defaultValue=0))
+        self.addParameter(QgsProcessingParameterNumber(self.TOLERANCE,
+                                                       self.tr('Tolerance'), minValue=0.0, maxValue=10000000.0, defaultValue=1.0))
 
-        self.addOutput(OutputVector(self.OUTPUT, self.tr('Simplified')))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT, self.tr('Simplified')))
+        self.addOutput(QgsProcessingOutputVectorLayer(self.OUTPUT, self.tr('Simplified')))
```

Comment:
@nyalldawson , I'm curious, what's the difference between the addParameter's FeatureSink parameter vs. the addOutput? How are these two related to one another?